File: src/com/company/BitManipulation/right_Most_Bit.java
package com.company.BitManipulation;

public class right_Most_Bit {
    public static void main(String [] args ){
      countRightMost(5);

    }
    static void countRightMost(int n){
        int result=1;
        while (n>0){
            if (n%2!=0){
                System.out.println(result);
                break;
            }
            n=n/2;
            result++;

        }

    }
}

File: src/com/company/BitManipulation/right_most_bit.kt
package com.company.BitManipulation
fun main(){


findRightMostBit(4)
    }
    fun findRightMostBit(k:Int) {

        var count = 1;
        var mask = 1;
        var n = k;
        while (n > 0) {

            if (n and mask == 1) {
                print(count)
                break
            }

            count++
            n = n shr 1

        }

    }